### DISCUSSION TITLE
Welcome to spateo-release Discussions!

### DISCUSSION THREAD

**Discussion by Xiaojieqiu** [aristoteleo/spateo-release#125] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nikita10293** [comment]
Hi, 

I am a computational Biologist. I recently started working on spatial transcriptomics data. Your work is a big support for us. 

I am interested in using spateo and dynamo for my analysis. However, I am stuck with two queries, described below. I request your guidance on the same. 

1. I am unable to generate a unspliced layer in count matrix/anndata from my bam file. My bam file has Cx, Cy, XF tags. However, Velocyto takes in CB tag for generating unspliced expression matrices. Could you please guide me on how to generate unspliced expression matrix, in such a case?

2. Also, I used spateo for cell binning, but the results lead to loss of some data. I am attaching the analysis images for your reference. Could you please guide me here as well. 
<img width="423" alt="Resultant_rna_image" src="https://user-images.githubusercontent.com/63791724/192920162-eb33a933-fce2-4cbb-b349-0f4104e7beb3.PNG">

<img width="200" alt="Spateo_img" src="https://user-images.githubusercontent.com/63791724/192920158-cf205ec9-d2f9-45ca-b747-15ef3d048b6a.PNG">

Requesting for your guidance. 

Thanks a ton. 
Nikita

**Reply by HailinPan** [reply]
Hi Nikita,

For the first question, did you map reads to genome reference by STAR? If so, I don’t think STAR would directly give a tag to delineate whether reads mapping to exon or intron regions. It would need extra-efforts to define whether a read is spliced or not through comparing the position it maps and gene mode information, such as gtf file.

For the second question, I guess it would be due to misalignment between staining image and RNA matrix. We recommend manually aligning staining image to RNA matrix before performing cell segmentation, sometimes you need flip, rotation, or scaling. And then you would get the same shape (height and width) of these two.

Best,
Hailin

**Comment by Danne1210** [comment]
Thanks for your wonderful framework that integrating multiple analysis for spatial data. Actually, I scanned another framework stereopy recently, and I noticed there are a few functions shared by the two framework, such as cell segmentation. Thus, I wondered what the similarities and differences between the two framework, especially the the shared function. Could you give some suggestion about it?

**Reply by Xiaojieqiu** [reply]
they are both developed independently, but there are some connections between the two projects previously. You can read about the preprint about Spateo here:  https://www.biorxiv.org/content/10.1101/2022.12.07.519417v1. Spateo is designed as a tool for more general framework for technologies other than Stereo-seq.

**Comment by Danne1210** [comment]
Thanks for your explanation about the relationships and differences of the two frameworks. I am going to read the preprint you provided about spateo and I believe it will help me in using the framework.

**Comment by uush99** [comment]
Dear Nikita,
Thanks for your wonderfull work, and recently, i meet a question,and I didn't get the answer. 
I have made a tissue mesh model named "eye_3d_mesh_model" through function ''st.tdr.construct_surface'', and in this function i used  one of parameters label : label = "eye". but now i want to change the label based on the mesh data made. and I don't want to use the  function "st.tdr.construct_surface" again, because I just want to change the label, and other data of the mesh don't need to change, so what should I do? could I just change the mesh model data, but I dont which function can do this?

my code as belows:

subpopulation_mesh, subpopulation_pc,_ = st.tdr.construct_surface(
        pc=embryo_pc,
        key_added="tissue",
        label='eye',#this i want to change to 'ear'
        alpha=1.0,
        cs_method="marching_cube",
        cs_args={"mc_scale_factor": 1.},
        smooth=10000,
        scale_factor=1
    )
ps: the "subpopulation_mesh" is data of pyvista.core.pointset.PolyData. How can I do for this ? Thanks for your time!
